js/modules/k6/data/data.go

```diff
@@ -144,3 +166,51 @@ func getShareArrayFromCall(rt *goja.Runtime, call goja.Callable) sharedArray {
 
 	return sharedArray{arr: arr}
 }
+
+// FIXME: not sure if this is needed. Do we want to expose the sharedArrayBuffer to users?
+// func (d *Data) sharedArrayBuffer(constructor goja.ConstructorCall) goja.Value {
+// 	// runtime := d.vu.Runtime()
+
+// 	// if d.vu.State() != nil {
+// 	// 	common.Throw(runtime, errors.New("new ImmutableArrayBuffer must be called in the init context"))
+// 	// }
+
+// 	// filename := constructor.Argument(0).String()
+// 	// if filename == "" {
+// 	// 	common.Throw(runtime, errors.New("empty filename provided to ImmutableArrayBuffer's constructor"))
+// 	// }
+
+// 	// // TODO: somehow acquire data? It looks like having a constructor just doesn't make sense.
+
+// 	// // FIXME: nil is a leftover, some data should actually make its way here? See previous TODO.
+// 	// array := d.immutable.get(filename, nil)
+// 	// return array.Wrap(runtime).ToObject(runtime)
+// 	// return array.Wrap(runtime).ToObject(runtime)
+// 	return nil
+// }
+
+func (i *sharedArrayBuffers) get(filename string, data []byte) SharedArrayBuffer {
+	i.mu.RLock()
+	array, exists := i.data[filename]
+	i.mu.RUnlock()
+	if !exists {
```

Comment:
There is a conception of the early returns, which I found pretty helpful because it also reduces nesting and improves the reading of the methods. 

For example:
```
if exists {
   return array
}
```

Do you think we can apply it here?

Reply:
Thanks for pointing that out. I really just copied it from `sharedArray`'s implementation. In principle, I would have prefered to keep it as a "negative" branching in this specific scenario, for (what feels to me as) clarity. But, in this specific scenario, I think it might actually lead to better performance to indeed check for existence instead. Considering the use case of this, I would expect most of the calls to this function to result in the data being already present in the slice. Hence, I assume (big assumption, I haven't verified that) it would make branch prediction's life easier if we were checking on a value that has 90% chances of being right, rather than the opposite.

Thanks for pointing it out :bear:

Reply:
Okay :)

Maybe clarify a bit. I'm not sure if we can gain some performance (since the variable value is calculated) here, but I was saying more from the readability perspective. 

For example:

```
if exists {
   return array
}

// ..some logic..

if another {
  ...
}

return array
```

is more straightforward (and easy to read) than

```
if ! exists {
   // ..some logic..

   if another {
       //  ...
       array = bar()
    }
}

return array
```

My issue with the second option is that since there is a single exit from the functions, you have to keep several conditions in mind (continue reading) before you recognize what is returning. But for sure, it's subjective, so just considering this as the sharing opinion.